rs/src/parse/expressions/literals.rs

```diff
@@ -677,32 +677,76 @@ fn parse_interval_year_to_month(
 /// Parses a day to second interval literal.
 fn parse_interval_day_to_second(
     x: &substrait::expression::literal::IntervalDayToSecond,
-    y: &mut context::Context,
+    ctx: &mut context::Context,
     nullable: bool,
     variations: Option<extension::simple::type_variation::ResolutionResult>,
 ) -> diagnostic::Result<Literal> {
-    proto_primitive_field!(x, y, days, |x, _| {
+    use substrait::expression::literal::interval_day_to_second::PrecisionMode;
+
+    proto_primitive_field!(x, ctx, days, |x, _| {
         if *x < -3650000 || *x > 3650000 {
             Err(cause!(
                 ExpressionIllegalLiteralValue,
-                "day count out of range -3_650_000 to 3_650_000"
+                "day count {} out of range -3_650_000 to 3_650_000",
+                *x
             ))
         } else {
             Ok(())
         }
     });
 
-    proto_primitive_field!(x, y, seconds);
-    proto_primitive_field!(x, y, microseconds);
+    proto_primitive_field!(x, ctx, seconds);
+    proto_primitive_field!(x, ctx, subseconds);
+    proto_field!(x, ctx, precision_mode);
```

Comment:
I think you can use:
```suggestion
    proto_required_field!(x, ctx, precision_mode);
```

Reply:
So I didn't require this because I'm not sure if we should require precision mode. In particular, requiring precision mode when no subseconds (or even seconds) are specified seems unnecessary; in fact, there's [a TPCH test](https://github.com/substrait-io/substrait-validator/blob/8b22bdbebccc27e38b0dafe73d5dae3f54fc78e1/tests/tests/tpc-h/tpc-h01.yaml#L265-L266) that uses `intervalDayToSecond: { days: 120 }` which has no precision mode set, but seems pretty reasonable since neither seconds nor subseconds are set.

Does that make sense? Does that seem reasonable?